test/persistence-hooks.suite.js

```diff
@@ -1374,6 +1374,90 @@ module.exports = function(dataSource, should) {
         });
       });
     });
+    
+    describe('PersistedModel.prototype.replace', function() {
+      it('triggers hooks in the correct order', function(done) {
+        monitorHookExecution();
+
+        existingInstance.replaceAttributes(
+          { name: 'replaced' },
+          function(err, record, created) {
+            if (err) return done(err);
+            triggered.should.eql([
+              'before save',
+              'persist',
+              'loaded',
+              'after save'
+            ]);
+            done();
+          });
+      });
+
+      it('aborts when `before save` hook fails', function(done) {
+        TestModel.observe('before save', nextWithError(expectedError));
+
+        existingInstance.replaceAttributes({ name: 'replaced' }, function(err) {
+          [err].should.eql([expectedError]);
+          done();
+        });
+      });
+
+      it('triggers `persist` hook', function(done) {
+        TestModel.observe('persist', pushContextAndNext());
+        existingInstance.replaceAttributes({ name: 'replacedName' }, function(err) {
+          if (err) return done(err);
+
+          observedContexts.should.eql(aTestModelCtx({
+            where: { id: existingInstance.id },
+            data: { name: 'replacedName' },
+            currentInstance: {
```

Comment:
You said not for `persist` though. I believe in [this comment](https://github.com/strongloop/loopback-datasource-juggler/pull/788#discussion-diff-47087458)